elements/additionalpower.lua

```diff
@@ -170,10 +202,28 @@ local function Visibility(self, event, unit)
 		end
 	end
 
-	if(shouldEnable) then
+	local isEnabled = element.__isEnabled
+
+	if(shouldEnable and not isEnabled) then
 		ElementEnable(self)
-	else
+
+		--[[ Callback: AdditionalPower:PostVisibility(isVisible)
+		Called after the element's visibility has been changed.
+
+		* self      - the AdditionalPower element
+		* isVisible - the current visibility state of the element (boolean)
+		--]]
+		if(element.PostVisibility) then
+			element:PostVisibility(true)
+		end
```

Comment:
This block ends with never calling Path, so the initial update when the element is shown never gets an update.

Reply:
`ElementEnable` is called in that block and it calls `Path`, which triggers the update.